plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-products-v2-controller.php

```diff
@@ -667,6 +743,28 @@ protected function api_get_related_ids( $product, $context ) {
 		return array_map( 'absint', array_values( wc_get_related_products( $product->get_id() ) ) );
 	}
 
+	/**
+	 * Get related products for a specific product.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @internal
+	 *
+	 * @phpstan-ignore-next-line missingType.generics -- WP_REST_Request generic type not specified, not practical for WP REST API.
```

Comment:
I noticed here `$request['id']` is not inferred, including it in the phpstan-param tells the expected type to phpstan

```
  Line   includes/rest-api/Controllers/Version2/class-wc-rest-products-v2-controller.php                       
 ------ ------------------------------------------------------------------------------------------------------ 
  :756   Parameter #1 $var of function intval expects array|bool|float|int|resource|string|null, mixed given.  
         🪪  argument.type      
```



```suggestion
	 * @phpstan-param WP_REST_Request<array<string>> $request
```